### PR TITLE
boards: add stm32f769 discovery

### PR DESCRIPTION
**Pull request by vincent-d**
This add basic support of stm32f769 discovery board and cpu (board manual [here ](http://www.st.com/content/ccc/resource/technical/document/user_manual/group0/7c/c9/c0/5f/4e/9b/44/eb/DM00276557/files/DM00276557.pdf/jcr:content/translations/en.DM00276557.pdf)). UART, LED and button works as expected, with SAUL support.

It also adds stm32f7 support in the RTC driver.

### CODE REVIEW

File: boards/stm32f7discovery/include/board.h [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#include "cpu.h"
+#include "periph_conf.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name LED pin definitions
+ * @{
```
Comment (aabadie) [inline boards/stm32f7discovery/include/board.h]: The doxygen comments look weird in this file.

File: boards/stm32f7discovery/include/board.h [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+#define LED0_PORT           GPIOJ
+#define LED1_PORT           GPIOJ
+#define LED2_PORT           GPIOA
+#define LED3_PORT           GPIOD
+#define LED0_MASK           (1 << 13)
+#define LED1_MASK           (1 << 5)
+#define LED2_MASK           (1 << 12)
+#define LED3_MASK           (1 << 4)
+
+#define LED0_ON             (LED_PORT->BSRR = LED0_MASK)
```
Comment (aabadie) [inline boards/stm32f7discovery/include/board.h]: LED_PORT => LED0_PORT ?
same comment for LED1 and LED2

File: boards/stm32f7discovery/include/board.h [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#define LED1_OFF            (LED_PORT->BSRR = (LED1_MASK << 16))
+#define LED1_TOGGLE         (LED_PORT->ODR  ^= LED1_MASK)
+
+#define LED2_ON             (LED_PORT->BSRR = LED2_MASK)
+#define LED2_OFF            (LED_PORT->BSRR = (LED2_MASK << 16))
+#define LED2_TOGGLE         (LED_PORT->ODR  ^= LED2_MASK)
+
+#define LED3_ON             (LED_PORT->BSRR = LED3_MASK)
+#define LED3_OFF            (LED_PORT->BSRR = (LED3_MASK << 16))
+#define LED3_TOGGLE         (LED_PORT->ODR  ^= LED3_MASK)
```
Comment (aabadie) [inline boards/stm32f7discovery/include/board.h]: If there are 4 LEDs why not adding the fourth one in board_init ?

**Comment by vincent-d** [comment]
Copy/paste are evil! 

@aabadie, comments addressed.

**Comment by vincent-d** [comment]
Fixed vendor header trailing white spaces and squashed.

**Comment by miri64** [comment]
Murdock issue most likely unrelated, see #7069.

**Comment by aabadie** [comment]
All green, go !

**Pull request merged**